srml/contract/src/rent.rs

```diff
@@ -0,0 +1,138 @@
+use crate::{BalanceOf, ContractInfo, ContractInfoOf, Module, TombstoneContractInfo, Trait};
+use runtime_primitives::traits::{As, Bounded, CheckedDiv, CheckedMul, Saturating, Zero};
+use srml_support::traits::{Currency, ExistenceRequirement, Imbalance, WithdrawReason};
+use srml_support::StorageMap;
+
+/// Evict and optionnaly pay rent, at block number. Return if evicted
```

Comment:
I think we need to mention explicitly that this function acts eagerly. This will connect this to the assumption written in exec.rs